packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -0,0 +1,217 @@
+    return {
+      __typename: KeyAgentType.Ledger,
+      accountIndex: this.#accountIndex,
+      communicationType: this.#communicationType,
+      extendedAccountPublicKey: this.#extendedAccountPublicKey,
+      knownAddresses: this.#knownAddresses,
+      networkId: this.networkId
+    };
+  }
+
+  static async getHidDeviceList(): Promise<string[]> {
+    return await TransportNodeHid.list();
+  }
+
+  static async createTransport({
+    communicationType,
+    activeTransport,
+    devicePath = ''
+  }: CreateTransportProps): Promise<TransportType> {
+    if (communicationType === CommunicationType.Node) {
+      return await TransportNodeHid.open(devicePath);
+    }
+    return await (activeTransport && activeTransport instanceof TransportWebHID
+      ? TransportWebHID.open(activeTransport.device)
+      : TransportWebHID.request());
+  }
+
+  static async createDeviceConnection(activeTransport: Transport): Promise<AppAda> {
+    const deviceConnection = new AppAda(activeTransport);
+    // Perform app check to see if device can respond
+    await deviceConnection.getVersion();
+    return deviceConnection;
+  }
+
+  static async establishDeviceConnection(communicationType: CommunicationType, devicePath?: string): Promise<AppAda> {
+    let transport;
+    try {
+      transport = await LedgerKeyAgent.createTransport({ communicationType, devicePath });
+      if (!transport || !transport.deviceModel) {
+        throw new TransportError('Transport failed');
+      }
+      const isSupportedLedgerModel = transport.deviceModel.id === 'nanoS' || transport.deviceModel.id === 'nanoX';
+      if (!isSupportedLedgerModel) {
+        throw new TransportError('Ledger device model not supported');
+      }
+      return await LedgerKeyAgent.createDeviceConnection(transport);
+    } catch (error) {
+      if (error.message.includes('cannot open device with path')) {
+        throw new TransportError('Connection already established', error);
+      }
+      // If transport is established we need to close it so we can recover device from previous session
+      if (transport) {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        transport.close();
+      }
+      throw error;
+    }
+  }
+
+  static async checkDeviceConnection(communicationType: CommunicationType, deviceConnection?: AppAda): Promise<AppAda> {
+    try {
+      if (!deviceConnection) {
+        return await LedgerKeyAgent.establishDeviceConnection(communicationType);
+      }
+      // Create / Check device connection with currently active transport
+      return await LedgerKeyAgent.createDeviceConnection(deviceConnection.transport);
+    } catch (error) {
+      // Device disconnected -> re-establish connection
+      if (error.name === 'DisconnectedDeviceDuringOperation') {
+        return await LedgerKeyAgent.establishDeviceConnection(communicationType);
+      }
+      throw error;
+    }
+  }
+
+  static async getXpub({
```

Comment:
:warning:
`KeyAgentBase` already has an abstract method `getExtendedAccountPublicKey`, so with the other changes to internalise the initialisation, it could be implemented as a non-static method.

Reply:
This should be private static then?

Reply:
Isn't this method implementing the intent of `getExtendedAccountPublicKey`?